### PR TITLE
Add storage class

### PR DESCRIPTION
**Pull request by magangloff**
Addressed some points pointed out by the TAs:
- Added a StorageHandler that stores and retrieves files. Currently only contains methods for the pictures but can be extended with methods for other files.
- Added an ImageHandler that handles the rotating of the images for the gallery.
- Added HttpClient for the download of the topography map

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/peakar/points/HttpClientTopographyMap.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    public HttpClientTopographyMap(Point point){
+        boundingBox = point.computeBoundingBox(BOUNDING_BOX_RANGE);
+        URL url = generateURL();
+
+        try {
+            HttpClient httpClient = HttpClientBuilder.create().build();
+            HttpRequestBase base = new HttpGet(url.toString());
+            HttpResponse response = httpClient.execute(base);
+
+            if (response.getStatusLine().getStatusCode() == 200) {
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/points/HttpClientTopographyMap.java:56]: Change `200` for a meaningful constant name

File: app/src/main/java/ch/epfl/sdp/peakar/points/HttpClientTopographyMap.java [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+     * @return Pair<int[][], Double> that contains the topographyMap and the mapCellSize
+     */
+    public Pair<int[][], Double> getTopographyMap() {
+           return result;
+    }
+
+    /**
+     * This method handles the parsing of the response obtained via HTTP request. It saves the
+     * size of the matrix to build and passes them to the buildMapGrid method.
+     *
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/points/HttpClientTopographyMap.java:79]: Add `@return` in javadoc

File: app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+     * Stores the bitmap on the device.
+     *
+     * @param context context of the application
+     * @param bitmap Bitmap that is to be stored
+     * @throws IOException thrown when bitmap could not be stores
+     */
+    public static void storeBitmap(Context context, Bitmap bitmap) throws IOException {
+        File screenshotFile = createPhotoFile(context);
+        FileOutputStream outputStream = new FileOutputStream(screenshotFile);
+        int quality = 100;
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java:33]: A comment explaining quality could be added here

File: app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        bitmap.compress(Bitmap.CompressFormat.JPEG, quality, outputStream);
+        outputStream.flush();
+        outputStream.close();
+    }
+
+    /**
+     * Retrieves all paths the images and sends back a list of them.
+     * @return list of all image paths.
+     */
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java:42]: Add `@param` to javadoc

File: app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+     * @return File where the photo will be stored
+     */
+    public static File createPhotoFile(Context context) {
+        return new File(getOutputDirectoryMedia(context),
+                new SimpleDateFormat(FILENAME_PHOTO, Locale.ENGLISH)
+                        .format(System.currentTimeMillis()) + PHOTO_EXTENSION);
+    }
+
+    /**
+     * Returns outpudirectory to store images. Use externel media if it is available, our app's
```
Comment (bgabriel998) [inline app/src/main/java/ch/epfl/sdp/peakar/utils/StorageHandler.java:68]:
```suggestion
     * Returns outpudirectory to store images. Use external media if it is available, our app's
```

**Review by bgabriel998 — CHANGES_REQUESTED** [review]
Minor changes requested

**Review by bgabriel998 — APPROVED** [review]

**Review by ravifrancesco — APPROVED** [review]
Good job, it seems more maintainable and easier to modify now

**Pull request merged**